luna/featurevis/transformations.py

```diff
@@ -73,33 +164,81 @@ def crop_or_pad(image, trans):
     if trans:
         trans_size = random.randint(-6, 6)
         image = tf.image.resize_with_crop_or_pad(
-            image, image.shape[1]-trans_size, image.shape[2]-trans_size)
+            image, image.shape[1] - trans_size, image.shape[2] - trans_size
+        )
     return image
 
 
-def vert_rotation(img, rotation):
-    """Randomly rotates the image.
+def vert_rotation(img, add_rotation):
+    """Rotating the image vertically.
 
     Args:
         img (list): image.
-        rotation (bool): True if image needs to rotate.
+        add_rotation (bool): True if image needs to rotate.
 
     Returns:
         img (list): rotated image.
     """
-    if rotation:
+    if add_rotation:
         img = tf.image.rot90(img)
     return img
 
 
-def random_flip(img, flip):
+def angle_conversion(angle, units):
+    """Converting the angle to the desirable format
+
+    Args:
+        angle (list): list of random angle values
+        units (str): conversion unit
+
+    Returns:
+        list: converted angle values to the desirable form
+    """
+    angle_f = tf.cast(angle, "float32")
+    if units.lower() == "degrees":
+        angle = 3.14 * angle_f / 180.0
+    elif units.lower() in ["radians", "rads", "rad"]:
+        angle = angle_f
+    return angle
+
+
+def rotation(img, *args, units="degrees", seed=None):
+    """Rotating the image with random angle
+
+    Args:
+        img (list): the image data
+        *args (tuple): list of arguments i.e. angles of rotatation
+        units (str, optional): the unit of rotation move. Defaults to "degrees".
+        seed ([type], optional): for reproducibility. Defaults to None.
+
+    Returns:
+        list: rotated image with selected values
+    """
+    angles = args[0]
+
+    if not isinstance(args[0], list):
+        angles = list(args)
+
+    if isinstance(angles, int):
+        angles = list(angles)
+
+    img = tf.convert_to_tensor(img)
+    angle = random_select(angles, seed=seed)
+    angle = angle_conversion(angle, units)
+    return tfa.image.rotate(img, angle)
+
+
+def flip(img, add_flip):
```

Comment:
Probably do not need 2nd arg.

Reply:
Resolved by giving transformations as function.